src/base/tests/test_util.py

```diff
@@ -42,6 +42,22 @@ def test_adapt_renamed_field(self):
 
         self.assertEqual(match_domain, new_domain)
 
+    @parametrize(
+        [
+            ("res.currency", [], "res.currency"),
+            ("res.currency", ["rate_ids"], "res.currency.rate"),
+            ("res.currency", ("rate_ids", "company_id"), "res.company"),
+            ("res.currency", ["rate_ids", "company_id", "user_ids"], "res.users"),
+            ("res.currency", ("rate_ids", "company_id", "user_ids", "partner_id"), "res.partner"),
+            ("res.users", ["partner_id"], "res.partner"),
+            ("res.users", ["nonexistent_field"], None),
+            ("res.users", ("partner_id", "removed_field"), None),
```

Comment:
Test an existing field that is not relational
```suggestion
            ("res.users", ("partner_id", "active"), None),
            ("res.users", ("partner_id", "removed_field"), None),
```

Reply:
Done, including a path over a non-relational field.